routing/segment.hpp

```diff
@@ -71,12 +79,108 @@ class Segment final
   bool m_forward = false;
 };
 
+class SegmentWeight final
+{
+public:
+  SegmentWeight() = default;
+
+  constexpr explicit SegmentWeight(double weight) : m_weight(weight) {}
```

Comment:
For wrappers around integral datatypes absence of explicit might be ok. For example, std::complex may be implicitly constructed from the double, and it simplifies it's usage and implementation.